cmake/pika_setup_apex.cmake

```diff
@@ -0,0 +1,10 @@
+# Copyright (c) 2022 ETH Zurich
+#
+# SPDX-License-Identifier: BSL-1.0
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
+# file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+if(PIKA_WITH_APEX)
+  pika_find_package(APEX REQUIRED)
```

Comment:
This is also a change that would be nice to have for other dependencies as well.